### DISCUSSION TITLE
[Question] Axioms for Theory of Lists

### DISCUSSION THREAD

**Discussion by frogfrog2** [prosyslab-classroom/cs424-program-reasoning#272] [Q&A]
Name: Jihwan Park

According to the 21st slide of lecture6, it seems that axioms of T_cons contain function congruences only for cons, car, and cdr. If this is correct, can we get f(a)=f(b) from the fact that a=b when we prove the example of the 22nd slide? I can't find any reason why axioms of T_cons don't just contain general function congruence, unlike axioms of theory of equality.

**Comment by KihongHeo** [comment]
Jihwan,
Very good question. $T_{cons}$ only contains `cons`, `car`, and `cdr`.
That is why on page 22, I used $T_{cons}^{=}$ which is the combination of two theories (list and equality).

**Comment by frogfrog2** [comment]
Thank you for your kind answer!
